### PR TITLE
Migrate `server` module tests to JUnit 5, part 8

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR finishes the migration of the `server` module to the new JUnit 5 test structure.

The following packages were migrated:
1. `tenant`
2. `transport`
3. `tuple`

Also some tests lay in the root `server` package itself, they were migrated too.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: server/src/test/java/io/spine/server/Given.java [hunk 1]
```diff
@@ -279,4 +297,96 @@ void event(CustomerCreated event) {
...
+        }
+
+        Throwable throwable() {
+            return throwable;
+        }
+
+        boolean isCompleted() {
+            return isCompleted;
+        }
+
```
Comment (armiol) [inline server/src/test/java/io/spine/server/Given.java]: This line is redundant.

File: server/src/test/java/io/spine/server/Given.java [hunk 2]
```diff
@@ -279,4 +297,96 @@ void event(CustomerCreated event) {
             getBuilder().mergeFrom(event.getCustomer());
         }
     }
+
+    /*
+     * QueryServiceTest environment.
```
Comment (armiol) [inline server/src/test/java/io/spine/server/Given.java]: Either 
```
`QueryServiceTest`
```
or `{@code QueryServiceTest}`.

Same below.

File: server/src/test/java/io/spine/server/ServerEnvironmentTest.java [hunk 3]
```diff
@@ -20,26 +20,35 @@
...
-        Tests.assertHasPrivateParameterlessCtor(ServerEnvironment.class);
+    @DisplayName(HAVE_PARAMETERLESS_CTOR)
+    void haveUtilityConstructor() {
+        assertHasPrivateParameterlessCtor(ServerEnvironment.class);
     }
 
     @Test
-    public void tell_when_not_running_under_AppEngine() {
+    @DisplayName("tell when not running under AppEngine")
+    void tellIfNotUnderGcloud() {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/ServerEnvironmentTest.java]: `tellIfNotInAppEngine`
G Cloud has a broader meaning (e.g. GCE).

File: server/src/test/java/io/spine/server/SubscriptionServiceTest.java [hunk 4]
```diff
@@ -307,40 +278,95 @@ public void cancel_subscription_on_topic() {
...
+    @Nested
+    @DisplayName("handle exceptions and call observer error callback for")
+    class HandleExceptionsOf {
+
+        @SuppressWarnings("ConstantConditions")
+        // As `null` is intentionally passed as a method param.
+        @Test
+        @DisplayName("subscription process")
+        void subscription() {
+            final BoundedContext boundedContext = setupBoundedContextWithProjectAggregateRepo();
```
Comment (armiol) [inline server/src/test/java/io/spine/server/SubscriptionServiceTest.java]: I see `final`s are still here. Maybe remove those?

File: server/src/test/java/io/spine/server/tenant/TenantRepositoryTest.java [hunk 5]
```diff
@@ -59,7 +62,8 @@ public void cache_passed_value() {
     }
 
     @Test
-    public void un_cache_values() {
+    @DisplayName("un-cache values")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/tenant/TenantRepositoryTest.java]: "Un-cache" is commonly called "evict from cache".

**Pull request merged**